components/sup/src/manager/periodic.rs

```diff
@@ -25,7 +25,7 @@ pub trait Periodic {
     /// When is the next time we should start a new task, given that
     /// we're going to start one right now?
     fn next_period_start(&self) -> SteadyTime {
-        SteadyTime::now() + TimeDuration::milliseconds(self.update_period())
+        SteadyTime::now() + TimeDuration::milliseconds(self.update_period() as i64)
```

Comment:
I'd suggest making `update_period()` return `Duration`, making this
```suggestion
        SteadyTime::now() + TimeDuration::from_std(self.update_period()).expect("out of range")
```

Reply:
Smart to use the `time` crate's `Duration` rather than stdlib. I like it.